### DISCUSSION TITLE
How can I download/save translations from a Project in a file?

### DISCUSSION THREAD

**Discussion by LoveKhatri** [crowdin/crowdin-api-client-js#223] [Q&A] [answered]
I have a crowdin project which I contributed to, and I want to download the translations from that project to use in the app.
- I would like to download the translations and save them in a json file.
I have tried most of the things, even the OTA client, but things didn't work out as intended.

The issue I am facing is getting the fileId to use in the below method
```js
translationsApi.buildProjectFileTranslation(
        projectId,
        fileId,
        {
            targetLanguageId: language
         }
        );
 ```
 if there is any method by which I can know which translations are completely done and download those, it would be much appreciated.

**Comment by dies** [comment, marked as answer]
One of the ways to get both `fileId` and its translation progress is to use the [Get Language Progress](https://developer.crowdin.com/api/v2/#operation/api.projects.languages.files.progress.getMany) API method. Like this: 

`client.translationStatusApi.withFetchAll().getLanguageProgress(220, 'fr');`

`withFetchAll()` is to get all results so you don't have to worry about pagination

'fr' is Crowdin language id,

220 is a project id.

The response you should expect is like following:

```json
{
  "data": [
    {
      "data": {
        "fileId": 12,
        "eTag": "fd0ea167420ef1687fd16635b9fb67a3",
        "words": {
          "total": 7249,
          "translated": 3651,
          "approved": 3637
        },
        "phrases": {
          "total": 3041,
          "translated": 2631,
          "approved": 2622
        },
        "translationProgress": 86,
        "approvalProgress": 86
      }
    }
  ],
  "pagination": {
    "offset": 0,
    "limit": 25
  }
}
```
